### DISCUSSION TITLE
原生 APP 和 H5 进行交互 window 对象被覆盖的问题

### DISCUSSION THREAD

**Discussion by yycmd** [NervJS/taro#12927] [Q & A]
### 这个特性解决了什么问题？
在原生 APP 和 H5 之间使用 WebViewJavaScriptBridge 框架注入双方的函数实现相互调用，由于 WebViewJavaScriptBridge  框架会将原生 APP 的函数挂载到 window 对象上，而 Taro 框架为了兼容各平台下的 window 对象而模拟了一个全新的，导致找不到通过 WebViewJavaScriptBridge  注入的原生APP函数

### 这个 API 长什么样？
在 config/index.js 的 h5 配置节点下增加保留标准的 window 对象配置 或者 import { getNative } from "@tarojs/taro" 等其他形式导出函数或属性去获取标准的 window 对象

**Comment by ZakaryCode** [comment]
h5 并没有模拟 window，该问题应当是其它原因导致的

**Comment by yycmd** [comment]
> h5 并没有模拟 window，该问题应当是其它原因导致的

这怎么判断是哪里出的问题

**Comment by ZakaryCode** [comment]
> 这怎么判断是哪里出的问题

可先检查是否由于原生 APP 函数加载时机导致

**Comment by yycmd** [comment]
> > 这怎么判断是哪里出的问题
> 
> 可先检查是否由于原生 APP 函数加载时机导致

APP 是对方开发的，我这边不是很方便去测试是否因为 APP 加载函数导致的

**Comment by ZakaryCode** [comment]
> APP 是对方开发的，我这边不是很方便去测试是否因为 APP 加载函数导致的

理论上需要 APP 方提供可验证环境，或者通过己方页面定位对应问题

1. 可验证空白 web 页面是否加载对应函数
2. 可验证函数加载具体时间（是否由于调用方法时函数未注入导致

> 该问题并不关联 issue 且并非新特性请求，将被转移到 Q & A discussions 继续
